crates/cast/src/script.rs

```diff
@@ -0,0 +1,360 @@
+        }?;
+
+        let result_end = buffer.ptr;
+        insert_value_to_cellref!(vm, output_start, result_start)?;
+        insert_value_to_cellref!(vm, output_end, result_end)?;
+
+        Ok(())
+    }
+}
+
+pub fn run(
+    script_path: &Utf8PathBuf,
+    provider: &JsonRpcClient<HttpTransport>,
+    runtime: Runtime,
+    path_to_scarb_toml: &Option<Utf8PathBuf>,
+) -> Result<ScriptResponse> {
+    check_compiler_path(true, Path::new(&script_path))?;
+
+    let metadata = parse_scarb_metadata(path_to_scarb_toml)?;
+
+    let corelib = corelib_for_package(&metadata, &metadata.workspace.members[0])?;
+    let db = &mut RootDatabase::builder().build()?;
+
+    init_dev_corelib(db, PathBuf::from(corelib));
+
+    let main_crate_ids = setup_project(db, Path::new(&script_path))?;
+    if DiagnosticsReporter::stderr().check(db) {
+        anyhow::bail!("failed to compile: {}", script_path);
+    }
+
+    let sierra_program = db
+        .get_sierra_program(main_crate_ids.clone())
+        .to_option()
+        .with_context(|| "Compilation failed without any diagnostics.")?;
+    let replacer = DebugReplacer { db };
+
+    let runner = SierraCasmRunner::new(
+        replacer.apply(&sierra_program),
+        Some(MetadataComputationConfig::default()),
+        OrderedHashMap::default(),
+    )
+    .with_context(|| "Failed setting up runner.")?;
+
+    let func = runner.find_function("::main")?;
+    let (entry_code, builtins) = runner.create_entry_code(func, &Vec::new(), usize::MAX)?;
+    let footer = runner.create_code_footer();
+    let instructions = chain!(
+        entry_code.iter(),
+        runner.get_casm_program().instructions.iter(),
+        footer.iter()
+    );
+
+    // import from cairo-lang-runner
+    let (hints_dict, string_to_hint) = build_hints_dict(instructions.clone());
+
+    // hint processor
+    let block_context = build_block_context(CheatnetBlockInfo::default());
+    let account_context = build_transaction_context();
+    let mut context = EntryPointExecutionContext::new(
+        &block_context.clone(),
+        &account_context,
+        ExecutionMode::Execute,
+    );
+
+    let mut blockifier_state =
+        CachedState::new(DictStateReader::default(), GlobalContractCache::default());
+    let mut execution_resources = ExecutionResources::default();
+
+    let syscall_handler = SyscallHintProcessor::new(
+        &mut blockifier_state,
+        &mut execution_resources,
+        &mut context,
+        // This segment is created by SierraCasmRunner
+        Relocatable {
+            segment_index: 10,
+            offset: 0,
+        },
+        CallEntryPoint::default(),
+        &string_to_hint,
+        ReadOnlySegments::default(),
+    );
+
+    let mut cairo_hint_processor = CairoHintProcessor {
+        blockifier_syscall_handler: syscall_handler,
+        hints: &string_to_hint,
+        provider,
+        runtime,
+    };
+
+    match runner.run_function(
+        runner.find_function("::main")?,
```

Comment:
We already have this function in `func` variable.

Reply:
Cairo-run does the same - ensure it's not some_long_name_ending_with_main I guess

Reply:
fixed